src/NATS.Client.Hosting/NatsHostingExtensions.cs

```diff
@@ -8,11 +8,24 @@ namespace NATS.Client.Hosting;
 public static class NatsHostingExtensions
 {
     /// <summary>
-    /// Add NatsConnection/Pool to ServiceCollection. When poolSize = 1, registered `NatsConnection` and `INatsCommand` as singleton.
-    /// Others, registered `NatsConnectionPool` as singleton, `NatsConnection` and `INatsCommand` as transient(get from pool).
+    /// Add NatsConnection/Pool to ServiceCollection. When poolSize = 1, registered `NatsConnection` and `INatsConnection` as singleton.
+    /// Others, registered `NatsConnectionPool` as singleton, `NatsConnection` and `INatsConnection` as transient(get from pool).
     /// </summary>
-    public static IServiceCollection AddNats(this IServiceCollection services, int poolSize = 1, Func<NatsOpts, NatsOpts>? configureOpts = null, Action<NatsConnection>? configureConnection = null)
+    public static IServiceCollection AddNats(
+        this IServiceCollection services,
+        int poolSize = 1,
+        Func<NatsOpts, NatsOpts>? configureOpts = null,
+        Action<NatsConnection>? configureConnection = null
+#if NET8_0_OR_GREATER
+        , string? key = null // This parameter is only available in .NET 8 or greater
```

Comment:
if you could suppress `Commas should not be preceded by whitespace` warnings we can get the build to be green (an unintentional check in native aot tests breaking the build for *any* earning)

Reply:
I added three more that I saw from the warnings on the build. Looks to have done the trick.